src/main/kotlin/org/pkl/lsp/services/DiagnosticsManager.kt

```diff
@@ -96,7 +98,8 @@ class DiagnosticsManager(project: Project) : Component(project) {
 
   fun getDiagnostics(module: PklModule): List<PklDiagnostic> {
     return project.cachedValuesManager.getCachedValue(
-      "DiagnosticsManager.getDiagnostics(${module.uri})"
+      "DiagnosticsManager.getDiagnostics(${module.uri})",
+      lock,
```

Comment:
Previously, this was using the cached values manager itself as the lock, which was the cause for deadlocks to arise (e.g. getDiagnostics -> getLibraryRoots -> getDiagnostics -> deadlock)